src/Swashbuckle.AspNetCore.SwaggerGen/Generator/OpenApiAnyFactory.cs

```diff
@@ -28,8 +29,14 @@ public static bool TryCreateFor(OpenApiSchema schema, object value, out IOpenApi
             else if (schema.Type == "number" && schema.Format == "double" && TryCast(value, out double doubleValue))
                 openApiAny = new OpenApiDouble(doubleValue);
 
-            else if (schema.Type == "string" && value.GetType().IsEnum)
-                openApiAny = new OpenApiString(Enum.GetName(value.GetType(), value));
+            else if (schema.Type == "string" && value.GetType().IsEnum) {
+                if (schema.Enum != null)
+                {
+                    var loweredValue = Enum.GetName(value.GetType(), value)?.ToLowerInvariant();
```

Comment:
consider instead:

```suggestion
                    var name = Enum.GetName(value.GetType(), value);
```

Reply:
Thanks!

Reply:
Anytime, hopefully https://github.com/isaacs/github/issues/1452 happens so that I could have provided a working suggestion without necessitating your further changes.